mods/ctf/ctf_modebase/bounties.lua

```diff
@@ -197,3 +257,60 @@ ctf_core.register_chatcommand_alias("put_bounty", "pb", {
 		return true, "Successfully placed a bounty of " .. amount .. " on " .. minetest.colorize(team_colour, player) .. "!"
 	end,
 })
+
+
+
+ctf_core.register_chatcommand_alias("bounty", "b", {
+	description = "Put bounty on someone's head using your score(10% fee)",
+	params = "<player> <score>",
+	func = function(name, params)
+		local bname, amount = string.match(params, "(.*) (.*)")
+		if not (amount and bname) then
+			return false, "Missing argument(s)"
+		end
+		amount = math.floor(amount)
+		local bteam = ctf_teams.get(bname)
+		if bteam == nil then
+			return false, "This player is either not online or not in any team"
+		end
+		if bteam == ctf_teams.get(name) then
+			return false, "You cannot put bounty on your teammate's head!"
+		end
+		if amount < 5 then
+			return false, "Sorry you must at least donate 15"
+		end
+		if amount > 100 then
+			return false, "Sorry but you cannot donate more than 100"
+		end
+
+		local current_mode = ctf_modebase:get_current_mode()
+		if not current_mode or not ctf_modebase.match_started then
+			return false, "Match has not started yet."
+		end
+		local cur_score = math.min(
+			current_mode.recent_rankings.get(name).score or 0,
+			(current_mode.rankings:get(name) or {}).score or 0
+		)
+		if amount > cur_score then
+			return false, "You haven't got enough to donate"
+		end
+		current_mode.recent_rankings.add(name, {score=-amount}, true)
+		if ctf_modebase.contributed_bounties[bname] == nil then
+			local contributors = {}
+			contributors[name] = amount
+			ctf_modebase.contributed_bounties[bname] = { total = amount, contributors = contributors }
+		else
+			if ctf_modebase.contributed_bounties[bname]["contributors"][name] == nil then
+				minetest.chat_send_all(name)
+				ctf_modebase.contributed_bounties[bname]["contributors"][name] = amount
+				ctf_modebase.contributed_bounties[bname]["total"] = ctf_modebase.contributed_bounties[bname]["total"] + amount
+			else
+				return false, "You cannot donate for someone's head twice"
+			end
+		end
+		minetest.chat_send_all(
+			minetest.colorize(
+			CHAT_COLOR,
+			string.format("%s donated %d for %s's head!", get_contributors(bname), amount, bname)))
```

Comment:
That wording is a little too bloodthirsty/PG-13

```suggestion
			string.format("%s placed a bounty of %d score on %s!", get_contributors(bname), amount, bname)))
```